src/Broadcast.chpl

```diff
@@ -135,4 +145,36 @@ module Broadcast {
     expandedVals = (+ scan expandedVals);
     return (expandedVals == 1);
   }
+
+  proc broadcast(segs: [?sD] int, segString: borrowed SegString, size: int) throws {
+    ref segOff = segString.offsets.a;
+    const high = sD.high;
+    var strSize: int;
```

Comment:
while i was looking into this i noticed we are missing the `broadcast` with permutation
 ```
proc broadcast(perm: [?D] int, segs: [?sD] int, segString: borrowed SegString) throws {
```
I feel like we should add that since it's possible for `permName` to be passed with a strings